torch/_decomp/decompositions.py

```diff
@@ -728,23 +728,37 @@ def slice_forward(
     start_val = start if start is not None else 0
     end_val = end if end is not None else sys.maxsize  # 2^63 - 1
 
-    if guard_size_oblivious(start_val < 0):
+    # negative indexing case
+    if guard_or_false(start_val < 0):
         start_val += sizes[dim]
-
-    if guard_size_oblivious(end_val < 0):
+    if guard_or_false(end_val < 0):
         end_val += sizes[dim]
 
-    if guard_size_oblivious(start_val < 0):
+    # negative out-of-bounds for start: treat as 0
+    if guard_or_false(start_val < 0):
```

Comment:
previously if start_val was -u0 this would be True. 
I wonder if we should some thing like
if start_val <=0 assumes its <0 
if start_val>=0 assume its >0 
if start_val between [-inf, inf] apply the reasoning we have in this PR (assume the positive)